streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamKTableJoin.java

```diff
@@ -46,21 +52,46 @@ public KStreamKTableLeftJoinProcessor(KTableValueGetter<K, V2> valueGetter) {
             this.valueGetter = valueGetter;
         }
 
-        @SuppressWarnings("unchecked")
         @Override
         public void init(ProcessorContext context) {
             super.init(context);
             valueGetter.init(context);
         }
 
         @Override
-        public void process(K key, V1 value) {
+        public void process(final K key, final V1 value) {
             // if the key is null, we do not need proceed joining
             // the record with the table
-            if (key != null) {
+            if (key != null && value != null) {
```

Comment:
ditto above: maybe add a comment explaining why we are excluding <key: null> for stream stream join.